plugins/Login/Security/LoginFromDifferentCountryDetection.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Login\Security;
+
+use Piwik\Container\StaticContainer;
+use Piwik\IP;
+use Piwik\Piwik;
+use Piwik\Plugins\GeoIp2\LocationProvider\GeoIp2;
+use Piwik\Plugins\Login\Emails\LoginFromDifferentCountryEmail;
+use Piwik\Plugins\Login\Model;
+use Piwik\Plugins\UserCountry\LocationProvider;
+use Piwik\Plugins\UsersManager\Model as UsersModel;
+
+class LoginFromDifferentCountryDetection
+{
+    /**
+     * @var Model
+     */
+    private $model;
+
+    /**
+     * @var UsersModel
+     */
+    private $usersModel;
+
+    /**
+     * @var array|null
+     */
+    private $location = null;
+
+    public function __construct(Model $model, UsersModel $usersModel)
+    {
+        $this->model = $model;
+        $this->usersModel = $usersModel;
+    }
+
+    public function isEnabled(): bool
+    {
+        // we need at least one GeoIP provider that is not the default or disabled one
+        return $this->isGeoIPWorking();
+    }
+
+    private function isGeoIPWorking(): bool
+    {
+        $provider = LocationProvider::getCurrentProvider();
+
+        return $provider instanceof GeoIp2
```

Comment:
I guess that makes sense.
I've added it as non-abstract method so that we don't need to add it to all providers we have and make extending location providers incompatible with the new class interface. By default we return false, and extending classes can override the method.